discover/portal/articles/210-setting-up/03-script-console/04-script-examples.markdown

```diff
@@ -0,0 +1,223 @@
+
+Since Liferay's script console does not provide an undo feature, it can be very
+convenient to set up a kind of preview mode. The purpose of a preview mode is to
+determine any permanent effects of a script before any information is actually
+saved to the Liferay database. The preview mode consists in using a
+`previewMode` flag which determines whether the operations with permanent
+effects should be executed or not. If `previewMode` is `true`, all of the data
+which would be permanently affected by the script is systematically printed.
+This providers the user executing the script with an outline of the data
+impacted by the script. If the user determines that everything is OK, the flag
+can be switched so that the script can make permanent updates to the database.
+
+Here's an example Groovy script that show the preview mode concept in action:
+
+    import java.util.Calendar
+    import com.liferay.portal.kernel.service.*
+    import com.liferay.portal.kernel.model.*
+    import com.liferay.portal.kernel.dao.orm.*
+    import static com.liferay.portal.kernel.workflow.WorkflowConstants.*
+
+    //
+    // Deactivate users never logged and created since more than 2 years
+    //
+
+    previewMode = true // Update this flag to false to really make changes
+
+    Calendar twoYearsAgo = Calendar.getInstance()
+    twoYearsAgo.setTime(new Date())
+    twoYearsAgo.add(Calendar.YEAR, -2)
+
+    DynamicQuery query = DynamicQueryFactoryUtil.forClass(User.class)
+            .add(PropertyFactoryUtil.forName("lastLoginDate").isNull())
+            .add(PropertyFactoryUtil.forName("createDate").lt(twoYearsAgo.getTime()))
+
+    users = UserLocalServiceUtil.dynamicQuery(query)
+
+    users.each { u ->
+             if(!u.isDefaultUser() && u.getStatus() != STATUS_INACTIVE) {
+                    out.println(u.getEmailAddress())
+                    if(!previewMode) {
+                            UserLocalServiceUtil.updateStatus(u.getUserId(), STATUS_INACTIVE)
+                    }
+             }
+    }
+
+    if(previewMode) {
+            out.println('Preview mode is on: switch off the flag and execute '
+                    + 'again this script to make changes to the database') 
+    }
+
+## Example 5: Plan a File Output for Long-Running Scripts [](id=example-5-plan-a-file-output-for-long-running-scripts)
+
+When a script has been running for a long time, it's possible for the script
+console to return an error even though the script can continue running and
+potentially conclude successfully. But it's impossible to know the outcome
+without the corresponding output!
+
+To bypass this limitation, you can send the output of the script console to a
+custom file instead of to the console itself or to the Liferay log. For example,
+consider this script:
+
+    import com.liferay.portal.kernel.service.*
+    import com.liferay.portal.kernel.dao.orm.*
+
+    // Output management
+
```

Comment:
We probably want to be careful here.  This shows the user that they can access the file system thru scripting console.  We should provide an explicit security warning.  Something like:

We strongly recommend only allowing access to Scrpiting Console to your portal administrator.  Scripting Console is meant for system operations and maintenance and not for end users.